src/melee.cpp

```diff
@@ -1406,6 +1410,13 @@ bool player::block_hit(Creature *source, body_part &bp_hit, damage_instance &dam
         melee_attack( *source, false, tec );
     }
 
+    // give stamina penalty for blocking, heavier weapons and arm encumbrance
+    // are more tiring, while higher block bonus means less effort expended
+    int encumbrance_cost = roll_remainder( ( encumb( bp_arm_l ) + encumb( bp_arm_r ) ) / 5.0f );
+    int weight_cost = weapon.weight() / ( 20 * std::max( 1, str_cur ) );
+    int mod_sta_block = ( weight_cost + encumbrance_cost + 20 - ( blocking_ability( shield ) / 2 ) ) * -1;
```

Comment:
Reasoning is twofold. One is to partially compensate for the item weight and encumbrance, the other is that effort needed to block an attack would decrease the less a shield has to move to cover that line of attack. Having a large scutum or riot shield (not an item in game) would mean a lot of angles of attack would already be covered.

Hence why I was uncertain about your suggestion that a dagger or fist would easily bypass a large shield. If anything, they would have to get closer to reach around the shield than if the attacker had a spear. Representing a zombie grabbing and clambering over/around the shield would likely be covered by the fact that the grab attack is countered by grab break techniques rather than block, along with the fact I can not find any code where blocking affects the results of a grab attempt.